### PR TITLE
Log all user actions

### PR DESCRIPTION
**Pull request by arkadyan**
Asana ticket: [Log all actions](https://app.asana.com/0/1199957631529739/1199969510486945)

• login
• search
• view document
• download document

### CODE REVIEW

File: test/support/helpers.ex [hunk 1]
```diff
@@ -15,4 +15,16 @@ defmodule Test.Support.Helpers do
       end)
     end
   end
+
+  defmacro reassign_log_level(level) do
```
[the remark this answers is not in the document]
Comment (arkadyan) [reply test/support/helpers.ex:19]: I was under the impression it did in order to reference `on_exit` in the scope of the calling describe block. But is there another way of handling that or something I'm getting wrong about that?

[the remark this answers is not in the document]
Comment (paulswartz) [reply test/support/helpers.ex:19]: I think you can call `on_exit` from a function? https://github.com/elixir-lang/elixir/blob/308255bda81e7f76f9bec838cef033e8e869981b/lib/ex_unit/lib/ex_unit/callbacks.ex#L351 shows that it uses the current process `self()`, which calling a function doesn't change.

File: lib/document_viewer_web/user_action_logger.ex [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  def log(username, action, params \\ %{}),
+    do: Logger.info("User action: username=#{username} action=#{action}#{params_string(params)}")
+
+  @spec params_string(map()) :: String.t()
+  defp params_string(params) do
+    Enum.reduce(
+      params,
+      "",
+      fn {key, val}, acc ->
+        "#{acc} #{key}=#{val}"
```
[the remark this answers is not in the document]
Comment (arkadyan) [reply lib/document_viewer_web/user_action_logger.ex]: Nice, thanks! I learned a couple of things from this.

File: test/support/helpers.ex (hunk 1 above)
[the remark this answers is not in the document]
Comment (arkadyan) [reply test/support/helpers.ex]: Yep, great, thanks. I think I confused myself by missing an import.

**Review by paulswartz — APPROVED** [review]
🍰

**Pull request merged**